### PR TITLE
Refs #32191 -- Added Signer.sign_object()/unsign_object().

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: docs/topics/signing.txt [hunk 1]
```diff
@@ -156,23 +182,49 @@ created within a specified period of time::
         otherwise raises ``SignatureExpired``. The ``max_age`` parameter can
         accept an integer or a :py:class:`datetime.timedelta` object.
 
+    .. method:: sign_object(obj, serializer=JSONSerializer, compress=False)
+
+        .. versionadded:: 3.2
+
+        Encode, optionally compress, sign complex data structure (e.g. list,
```
Comment (apollo13) [inline docs/topics/signing.txt:189]: Nitpicking, but maybe add an "add" here? "…, and sign complex data"

File: docs/topics/signing.txt [hunk 2]
```diff
@@ -156,23 +182,49 @@ created within a specified period of time::
...
 
 If you wish to protect a list, tuple or dictionary you can do so using the
-signing module's ``dumps`` and ``loads`` functions. These imitate Python's
-pickle module, but use JSON serialization under the hood. JSON ensures that
-even if your :setting:`SECRET_KEY` is stolen an attacker will not be able
-to execute arbitrary commands by exploiting the pickle format::
+``Signer.sign_object()`` and ``unsign_object()`` methods, or signing module's
+``dumps()`` or ``loads()`` functions (which are shortcuts for
+``TimestampSigner(salt='django.core.signing').sign_object()/unsign_object()``).
+These imitate Python's pickle module, but use JSON serialization under the
```
Comment (apollo13) [inline docs/topics/signing.txt:213]: I'd drop the comments about pickle and just say that serialization (by default) is done via JSON. No strong feeling about it though.

**Review by apollo13 — APPROVED** [review]

File: docs/topics/signing.txt (hunk 1 above)
Comment (felixxm) [reply docs/topics/signing.txt:189]: There is _"and"_ before the last _", and append current ..."_.

File: docs/topics/signing.txt (hunk 2 above)
Comment (felixxm) [reply docs/topics/signing.txt:213]: Agreed.

**Comment by felixxm** [comment]
@apollo13 Thanks for checking edits :+1:

File: docs/topics/signing.txt (hunk 1 above)
Comment (apollo13) [reply docs/topics/signing.txt:189]: ha, reading comprehension skills… But then the ordering is wrong. Signing is always last (ie the timestamp is also signed!)

Comment (felixxm) [reply docs/topics/signing.txt]: Good catch :dart: , updated.

**Pull request merged**